src/test/java/seedu/sellsavvy/model/order/DateTest.java

```diff
@@ -20,24 +22,52 @@ public void constructor_invalidDate_throwsIllegalArgumentException() {
     }
 
     @Test
-    public void isValidDate() {
+    public void isValidDateRegex() {
         // null date value
-        assertThrows(NullPointerException.class, () -> Date.isValidDate(null));
+        assertThrows(NullPointerException.class, () -> Date.isValidDateRegex(null));
+
+        // invalid date regex
+        assertFalse(Date.isValidDateRegex("")); // empty string
+        assertFalse(Date.isValidDateRegex(" ")); // spaces only
+        assertFalse(Date.isValidDateRegex("date")); // not a date
+        assertFalse(Date.isValidDateRegex("2-02-2002")); // invalid day format
```

Comment:
Can this be renamed to `single digit day`?

Reply:
Updated in commit.